### ISSUE TITLE
Error: The package 'react-native-android-widget' doesn't seem to be linked

### ISSUE THREAD

**Issue by vaporrrr** [sAleksovski/react-native-android-widget#12]
When running the app on iOS (expo bare workflow) with this package installed, this error occurs:
```
 ERROR  Error: The package 'react-native-android-widget' doesn't seem to be linked. Make sure: 

- You rebuilt the app after installing the package
- You are not using Expo Go
, js engine: hermes
 ERROR  Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes
```
I understand the library is for adding widgets on android, so therefore iOS support is not necessarily implied, but for cross platform support it would be nice for the package to simply do nothing on iOS and for the rest of the app to still work.

**Comment by peerdavid** [comment]
I get the same error, but for Android (Expo):
```
 ERROR  Error: The package 'react-native-android-widget' doesn't seem to be linked. Make sure: 

- You rebuilt the app after installing the package
- You are not using Expo Go
, js engine: hermes
```

**Comment by vaporrrr** [comment]
Android is working fine for me with expo 48.0.11

**Comment by sufiyanleanspace** [comment]
Any update on this issue?

**Comment by peerdavid** [comment]
> Android is working fine for me with expo 48.0.11

I used 48.0.6 and tested now 48.0.11, but still get this error. I followed the tutorial but could easily be that I missed something as I'm definitley not a Expo/React expert :/

**Comment by sufiyanleanspace** [comment]
I tried running the example-expo project of this repo does not seems to be working, Now I am trying the example project which is without the expo.

**Comment by sufiyanleanspace** [comment]
in example I am facing this issue:

`* What went wrong:
Could not open settings generic class cache for settings file '/Users/Sufiyan/react-native-android-widget/example/android/settings.gradle' (/Users/Sufiyan/.gradle/caches/7.5.1/scripts/5hkoeqehrsuwg7ysrn5djf1yv).
> BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 63

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org`

**Comment by sAleksovski** [comment]
@vaporrrr 
I just released `0.4.1`.
I don't have a Mac device, so I cannot test it for iOS, but I think it should be fixed now.
Can you verify it?

@peerdavid 
You should not get the same error on Android (Expo).
Did you create a new Dev Client after adding the library? The library does not work in Expo Go.
The `example-app.apk` in the [release files](https://github.com/sAleksovski/react-native-android-widget/releases) is the `example-expo` app from the repo, so it works with Expo.

@sufiyanleanspace 
I don't think your issue is related to this.
I found this https://github.com/react-native-community/cli/issues/1719 , it is probably related to your gradle errors.
As for running the `example-expo`, you first need to build a Dev Client for it, since it does not work with Expo Go.
The example-expo app references the library in `package.json` by a local file. You can get it by running `yarn build:lib`.

**Comment by vaporrrr** [comment]
Thanks, looks like it works on iOS now, ill let you know if there are any other issues, but I think this issue can be closed.

**Comment by sufiyanleanspace** [comment]
@sAleksovski Thanks for the explanation it is working now, but the documentation for that is probably worth creating, I can give it a try if you don't mind.

**Comment by sAleksovski** [comment]
@sufiyanleanspace It is mentioned in the docs (https://saleksovski.github.io/react-native-android-widget/docs/tutorial/register-widget-expo#build-dev-client) that you need to create a Dev Client for Expo.
If you run `yarn build:dev-client` it will automatically build the library.

**Comment by sufiyanleanspace** [comment]
Thank you!

**Comment by peerdavid** [comment]
> @vaporrrr I just released `0.4.1`. I don't have a Mac device, so I cannot test it for iOS, but I think it should be fixed now. Can you verify it?
> 
> @peerdavid You should not get the same error on Android (Expo). Did you create a new Dev Client after adding the library? The library does not work in Expo Go. The `example-app.apk` in the [release files](https://github.com/sAleksovski/react-native-android-widget/releases) is the `example-expo` app from the repo, so it works with Expo.
> 
> @sufiyanleanspace I don't think your issue is related to this. I found this [react-native-community/cli#1719](https://github.com/react-native-community/cli/issues/1719) , it is probably related to your gradle errors. As for running the `example-expo`, you first need to build a Dev Client for it, since it does not work with Expo Go. The example-expo app references the library in `package.json` by a local file. You can get it by running `yarn build:lib`.

Thank you so much for your help and this great work!
